apps/encryption/lib/Crypto/Crypt.php

```diff
@@ -107,6 +115,7 @@ public function __construct(ILogger $logger, IUserSession $userSession, IConfig
 		$this->config = $config;
 		$this->l = $l;
 		$this->supportLegacy = $this->config->getSystemValueBool('encryption.legacy_format_support', false);
+		$this->useLegacyEncoding = $this->config->getSystemValueBool('encryption.use_legacy_encoding', false);
```

Comment:
Seems better to store the encoding format directly to be more explicit of what legacy means.
If adopted, we'll need to adapt the logic and nomenclature in other places.

```suggestion
		$this->encodingFormat = $this->config->getSystemValueBool('encryption.encryption_encoding_format', self::DEFAULT_ENCODING_FORMAT);
```

Reply:
We could certainly change this from a boolean to a string with allowed values "base64" and "binary", but as I said I'd leave the base64-encoded files without a header, for compatibility.

On the other hand calling base64 "legacy" has the nice side effect of making clear that there's no reason to use it, unless for very specific reasons (and I can only think of compatibility with older releases honestly).

Reply:
I renamed the variables, but the default encoding format is binary with these patches because `encryption.use_legacy_encoding`/`encryption.use_legacy_base64_encoding` defaults to `false` so I'd leave `const DEFAULT_ENCODING_FORMAT` unchanged.

Reply:
Thanks for the rename :)

Regarding the default topic, I was talking about the value of `encoding` in the file's headers. When there is no `encoding` header, it defaults to base64. And the fact that `use_legacy_base64_encoding` defaults to `false` make it indeed looks like it is the other way. I would be more comfortable if the two contexts were aligned.

Reply:
Oh, yes, now I understand. I pushed a new version of the rename patch that also changes `DEFAULT_ENCODING_FORMAT` to `BINARY_ENCODING_FORMAT` and it's certainly clearer.

Reply:
Thanks a lot @plumbeo